File: Week_02/Practicum/Majority Element II.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        int count = 1;
        int size = nums.size();
        vector<int> result;

        for (int i = 1; i < size; i++) {
            if(nums[i] == nums[i-1])
                count++;
            else {
                if(count > size /3 )
                    result.push_back(nums[i-1]);
                count = 1;
            }
        }

        if (count > size / 3) {
            result.push_back(nums[size - 1]);
        }

        return result;
    }
};

File: Week_02/Practicum/python/Majority Element II.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        nums.sort()
        n = len(nums)
        v = []
        i = 0
        while i + n // 3 < n:
            if nums[i] == nums[i + n // 3] and nums[i] not in v:
                v.append(nums[i])
            i += 1

        return v